### DISCUSSION TITLE
How intelligently share chakra in monorepo architecture

### DISCUSSION THREAD

**Discussion by bit-crafter** [chakra-ui/chakra-ui#5233] [Q&A]
I have  2 different dashboards using chakra UI with the same theme configuration. I want to create another workspace like UI components to store custom chakra components. So the final structure  will be

```
root
  dashboard-1
  dashboard-2
  ui-components
  package.json
```

what is the recommended way to use the same configuration file, share custom chakra components, and use shared ChakraProvider?

**Comment by Rykuno** [comment]
Here is an example if you wish to copy its structure. https://github.com/NoQuarterTeam/boilerplate/tree/main/packages

**Comment by nivo33** [comment]
Would be happy for an update on this. Currently, if using a shared package in a monorepo, there is no way to share the ChakraProvider, which means every component exported from the shared UI package needs to be wrapped in its own provider which is quite a headache, and doesn't allow things like setting global fontSizes once
